staging/src/k8s.io/apiserver/pkg/server/resourceconfig/helpers.go

```diff
@@ -141,22 +167,43 @@ func MergeAPIResourceConfigs(
 		if err != nil {
 			return nil, err
 		}
-		if enabled {
+
+		switch len(tokens) {
+		case 2:
+			versionPreferences = append(versionPreferences, versionEnablementPreference{
+				key:          key,
+				enabled:      enabled,
+				groupVersion: groupVersion,
+			})
+		case 3:
```

Comment:
non-blocking: since we're not catching typos here, should we do a sanity check like making sure the resource name is all-lowercase? I expect "$group/$version/$Kind" and "$group/$version/$singular_resource" to be the most common typos

Reply:
> non-blocking: since we're not catching typos here, should we do a sanity check like making sure the resource name is all-lowercase and warn if it looks wrong? I expect "$group/$version/$Kind" and "$group/$version/$singular_resource" to be the most common typos

I added this.